packages/dara-core/dara/core/configuration.py

```diff
@@ -52,6 +52,7 @@ class Configuration(GenericModel):
     """Definition of the main framework configuration"""
 
     auth_config: BaseAuthConfig
+    registry_lookup: Dict[str, Callable]
```

Comment:
If we're going with the dict approach, could we define an enum for the registry names which we currently support for the custom registry lookup?

e.g. in the registries file have an enum

```python
class RegistryType(str, Enum):
   FOO = 'Foo'
   BAR = 'Bar'
   BAZ = 'Baz'
```

Then in registry lookup we can define which registries we support custom lookup for; otherwise it's not clear which ones work

```python
RegistryLookupKey = Literal[RegistryType.FOO, RegistryType.BAR]
CustomRegistryLookup = Dict[RegistryLookupKey, Callable[<some coroutine type>]]
```

This way we can specify which handlers are supported for the registry lookup

```python
def add_registry_lookup(self, registry_lookup: CustomRegistryLookup):
```